### ISSUE TITLE
[java] UselessParentheses: False positive in ternary else expression

### ISSUE THREAD

**Issue by Vampire** [pmd/pmd#6274]
**Affects PMD Version:** 7.18.0

**Rule:** [UselessParentheses](https://docs.pmd-code.org/pmd-doc-7.18.0/pmd_rules_java_codestyle.html#uselessparentheses)

**Description:**

Given the below code sample, both of the first two `bae` lines are hard to read and need some operator in-head parsing to see the outcome.

The second two `bae` lines are imho greatly preferred to increase readability.

With `ignoreClarifying` enabled, the last line is indeed accepted, but the next to last line is complained about to have useless parentheses.

If that is a measure for you, IntelliJ also makes the second two `bae` lines out of the first two `bae` lines if you ask it to add clarifying parentheses.

**Code Sample demonstrating the issue:**

```java
boolean foo, bar, baz, bam, boo, baa;
foo = bar = baz = bam = boo = new Random().nextBoolean();

baa = foo ? bar : baz ? bam : boo;
baa = foo ? bar ? baz : bam : boo;

baa = foo ? bar : (baz ? bam : boo);
baa = foo ? (bar ? baz : bam) : boo;
```

**Expected outcome:**

PMD reports a violation at line 7 but that's wrong. That's a false positive.

**Running PMD through:** *Gradle*

**Comment by oowekyala** [comment]
Mm yes maybe we could consider the parentheses in the next to last line to be clarifying too. This would be consistent with how we treat the last line.
